### DISCUSSION TITLE
Combining randomized smoothing with existing classifier/model

### DISCUSSION THREAD

**Discussion by DmitriiGudin** [Trusted-AI/adversarial-robustness-toolbox#1400] [Q&A]
Hello friends,

I am working on a course project in which I am trying to combine the XGBoost classifier (trained, for instance, on the MNIST data) with randomized smoothing to achieve provable certifiable robustness. I am new to ART and am having trouble figuring out the syntax.

I have a base trained classifier _clf_ obtained as follows:

```
def train_base_model (train_X, train_Y, test_X, test_Y): # Trains and returns an XGBoost model. train_X must be an array of 1-dimensional arrays, and train_Y is an array of integers corresponding to different labels.
    parameters = {"objective": "multi:softprob", "metric": "accuracy", "num_class": len(params.classes)}
    dtrain = xgb.DMatrix(train_X, np.argmax(train_Y, axis=1))
    dtest = xgb.DMatrix(test_X, np.argmax(test_Y, axis=1))
    evals = [(dtest, "test"), (dtrain, "train")]
    model = xgb.train(params=parameters, dtrain=dtrain, num_boost_round=50, evals=evals)
    clf = XGBoostClassifier(model=model, clip_values=(min_pixel_value, max_pixel_value), nb_features=params.N_features, nb_classes=len(params.classes))
    return clf
```

Now, I tried using the _RandomizedSmoothingMixin_ wrapper from _art.estimators.certification.randomized_smoothing_ to combine it either with the classifier _clf_, or the base model _model_, from the function above - but could not figure out how to do it. Reading this [page](https://adversarial-robustness-toolbox.readthedocs.io/en/latest/modules/estimators/certification_randomized_smoothing.html) still left me confused.

Any tips are appreciated. Thanks!

**Comment by beat-buesser** [comment]
Hi @DmitriiGudin I think it's an interesting project. You would have to create a similar estimator as for example [`PyTorchRandomizedSmoothing`](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/main/art/estimators/certification/randomized_smoothing/pytorch.py) but for XGBoost. This file also shows how to use `RandomizedSmoothingMixin`.
